packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFInstanceManager.h

```diff
@@ -25,34 +38,33 @@ NS_ASSUME_NONNULL_BEGIN
  * @param instance The instance to be stored.
  * @param instanceIdentifier The identifier to be paired with instance. This value must be >= 0.
  */
-- (void)addInstance:(NSObject *)instance withIdentifier:(long)instanceIdentifier;
+- (void)addFlutterCreatedInstance:(NSObject *)instance withIdentifier:(long)instanceIdentifier;
 
 /**
- * Removes the instance paired with a given identifier from the manager.
- *
- * @param instanceIdentifier The identifier paired to an instance.
+ * Adds a new instance that was instantiated by the host platform.
  *
- * @return The removed instance if the manager contains the given instanceIdentifier, otherwise
- * nil.
+ * @param instance The instance to be stored.
+ * @return The randomly generated identifier stored with instance.
  */
-- (nullable NSObject *)removeInstanceWithIdentifier:(long)instanceIdentifier;
+- (long)addHostCreatedInstance:(nonnull NSObject *)instance;
 
 /**
- * Removes the instance from the manager.
+ * Removes `instanceIdentifier` and its associated strongly referenced instance, if present, from
+ * the manager.
  *
- * @param instance The instance to be removed from the manager.
+ * @param instanceIdentifier The identifier paired to an instance.
  *
- * @return The identifier of the removed instance if the manager contains the given instance,
- * otherwise NSNotFound.
+ * @return The removed instance if the manager contains the given instanceIdentifier, otherwise
+ * nil.
  */
-- (long)removeInstance:(NSObject *)instance;
+- (nullable NSObject *)removeStrongReferenceWithIdentifier:(long)instanceIdentifier;
```

Comment:
That makes sense. My thoughts were that the instance could still be accessible in the manager because of the weak reference. Some other object, outside or inside the manager, could hold a reference to it. Like how the `WebView` may hold on to the `WKNavigationDelegate`.